src/ImageSharp/Formats/Gif/GifEncoderCore.cs

```diff
@@ -170,88 +227,418 @@ public void Encode<TPixel>(Image<TPixel> image, Stream stream, CancellationToken
         }
     }
 
-    private void EncodeFrame<TPixel>(
+    private void EncodeFirstFrame<TPixel>(
+        Stream stream,
+        GifFrameMetadata? metadata,
+        IndexedImageFrame<TPixel> quantized,
+        int transparencyIndex)
+        where TPixel : unmanaged, IPixel<TPixel>
+    {
+        this.WriteGraphicalControlExtension(metadata, transparencyIndex, stream);
+
+        Buffer2DRegion<byte> region = ((IPixelSource)quantized).PixelBuffer.GetRegion();
```

Comment:
Nit: do we need the region here? I don't see us slicing down the full buffer.

Reply:
`WriteImageData` requires a region as we slice down the buffer. I'll rewrite it though so it accepts a buffer and a rectangle.